app/Http/Controllers/OfficerController.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\OfficerRequest;
+use App\Models\Officer;
+use Illuminate\Support\Str;
+
+class OfficerController extends Controller
+{
+    public function index()
+    {
+        return view('pages.officers.index',[
+            'officers' => Officer::paginate(20)
+        ]);
+    }
+
+    public function create()
+    {
+        return view('pages.officers.create',[
+            'positions' => Officer::getPositions()
+        ]);
+    }
+
+    public function store(OfficerRequest $request)
+    {
+        if(!Officer::isValidPosition($request->position))
+        {
+            return back()->withErrors(['position' => 'Invalid Position'])->withInput(['fullname' => $request->fullname]);
+        }
+
+        Officer::create([
+            'fullname' => Str::title($request->fullname),
+            'position' => $request->position
+        ]);
+
+        return redirect(route('admin.officers.index'))->with([
+            'created' => true,
+            'message' => 'New Officer has been successfully added!'
+        ]);
+    }
+
+    public function edit(Officer $officer)
+    {
+        $officer = Officer::findOrFail($officer->id);
+
+        return view('pages.officers.edit',[
+            'officer' => $officer,
+            'positions' => Officer::getPositions()
+        ]);
+    }
+
+    public function update(OfficerRequest $request, Officer $officer)
+    {
+        if(!Officer::isValidPosition($request->position))
+        {
+            return back()->withErrors(['position' => 'Invalid Position'])->withInput(['fullname' => $request->fullname]);
+        }
+
+        $officer = Officer::findOrFail($officer->id);
+
+        $officer->fullname = Str::title($request->fullname);
+        $officer->position = $request->position;
+        $officer->save();
+
+        return redirect(route('admin.officers.index'))->with([
+            'created' => true,
+            'message' => 'Officer has been successfully updated!'
+        ]);
+    }
+
+    public function destroy(Officer $officer)
+    {
+        $officer = Officer::findOrFail($officer->id);
```

Comment:
You don't have to findOrFail here. Route model binding already do this for us, by expecting it in your parameter. You can just directly call `$officer->delete()`

Reply:
Like what you suggest, I remove the `Officer::findOrFail()`.
**Image evidence**
![image](https://user-images.githubusercontent.com/43285621/134672432-cef4e356-585a-4f8f-85bd-89a23e28c70e.png)

And this code works perfectly fine. Just because I'm having a trust issue so I tried to run the test and it really works. Thank you for sharing your knowledge :D